Release: Emulsify 7.x

callinmullaney opened the pull request:
**This PR does the following:**

- Prepares Emulsify Drupal for the 7.x release line with Drupal `^11.3 || ^12` support.
- Removes the Stable9 base theme dependency and adds parent-owned template coverage for Drupal core template parity.
- Updates the Emulsify parent theme to use `emulsify_tools:^2.0` and removes the direct `drupal/components` dependency.
- Updates Whisk to remain a hidden, generation-only starterkit source while generated child themes use `emulsify` as their runtime parent theme.
- Upgrades the generated Whisk starter theme to Emulsify Core `^4.0.0` and replaces Webpack references with the Vite-based build flow.
- Adds generated favicon package support to theme settings, including SVG upload, preview behavior, platform-specific browser/iOS/Android settings, manifest generation, and page-level head tag attachments.
- Adds favicon defaults, schema entries, admin preview CSS/JS, generation services, and smoke tests for package generation/portability.
- Refactors legacy hook includes into Drupal 11.3-style hook classes using `#[Hook]` attributes.
- Adds readiness tooling for render parity, no-Stable9 checks, starterkit generation, reference-page rendering, normalized HTML comparison, and release validation.
- Updates browser support config, npm package metadata, Husky setup, and semantic-release dependencies for the 7.0.0 release.

### Related Issue(s)

- N/A

### Notes:

- This is a major release PR.
- Drupal 10 support is removed.
- Whisk should not be enabled directly; it is only used as the starterkit source for generated themes.
- Favicon generation requires the PHP extensions needed to rasterize SVG assets, including Imagick and GD.

### Functional Testing:

- [ ] Install this branch in a clean Drupal `11.3+` site.
- [ ] Confirm Composer resolves `drupal/core:^11.3 || ^12` and `drupal/emulsify_tools:^2.0`.
- [ ] Enable the Emulsify theme and confirm the site renders without Stable9 as a base theme.
- [ ] Generate a child theme from Whisk and confirm the generated theme uses `emulsify` as its base theme.
- [ ] In the generated child theme, run `npm install`, `npm run build`, and `npm run develop`.
- [ ] Confirm Vite-generated assets are created and Storybook still starts successfully.
- [ ] Configure a generated favicon package from theme settings using an SVG source.
- [ ] Confirm favicon assets, manifest output, and related head tags are generated and attached to the page.
- [ ] Confirm the favicon preview UI updates in the theme settings form.
- [ ] Run the readiness/smoke scripts or GitHub Actions checks for template parity, no-Stable9 readiness, starterkit generation, and favicon generation.

### Security

- SVG favicon uploads are sanitized before package generation.
- Unsafe SVG nodes, event handlers, and external references are stripped from the stored/generated favicon source.
- No other known security concerns.

### Accessibility

- Parent-owned template coverage is intended to preserve Drupal core template behavior while removing the Stable9 runtime dependency.
- Favicon and build-tooling changes should be visually/semantically neutral.

Commit:
feat: replace webpack references with Vite and upgrade emulsify core version

Commit:
feat: vite plugins project extensions file

josue2591 commented on whisk/package.json:
```diff
@@ -38,9 +37,9 @@
     "tokens:transform": "token-transformer ./src/tokens/figma.tokens.json ./src/tokens/transformed.tokens.json",
     "twatch": "jest --no-coverage --watch --verbose",
-    "webpack": "webpack --watch --config node_modules/@emulsify/core/config/webpack/webpack.dev.js"
+    "vite": "vite build --watch --config node_modules/@emulsify/core/config/vite/vite.config.js",
```
Remove comma
```suggestion
    "vite": "vite build --watch --config node_modules/@emulsify/core/config/vite/vite.config.js"
```

callinmullaney committed:
Apply suggestion from @josue2591

Co-authored-by: josue2591 <josuegv2591@gmail.com>

Commit:
feat: use bundle classes for hooks

Commit:
feat(theme): own baseline parent theme templates

Commit:
feat(starterkit): add whisk as a starter source theme

Commit:
ci(theme): add stable9 readiness and starterkit smoke checks

Commit:
chore(metadata): align 6.x support metadata to Drupal 10.3+

Commit:
docs(readme): document child theme generation workflows

Commit:
fix: refine setup and startkit github actions test

Commit:
chore(release): drop Drupal 10.3 support and align metadata

Commit:
refactor(theme): migrate theme hooks to Drupal hook attributes

Commit:
feat(theme): add generated favicon package workflow

callinmullaney committed:
Merge branch '7.x' into release-7

callinmullaney committed:
Merge pull request #351 from emulsify-ds/release-7

Release 7

Commit:
fix(theme): hide favicon reset action until a package exists

Commit:
chore(deps): remove drupal/components dependency declarations

callinmullaney committed:
Merge pull request #352 from emulsify-ds/release-7

Release 7

Commit:
chore: remove components dependency from whisk

callinmullaney committed:
Merge pull request #353 from emulsify-ds/release-7

chore: remove components dependency from whisk

Commit:
feat(theme): refine generated favicon workflow

Commit:
chore(release): add release readiness checks

callinmullaney committed:
Merge pull request #354 from emulsify-ds/release-7

Release 7

Commit:
feat: drop stable9 support and port over templates

Commit:
chore: test commit tests

Commit:
chore: refine husky commit test

Commit:
refactor(theme): use add_attributes across templates

callinmullaney committed:
Merge branch '7.x' into release-7

callinmullaney committed:
Merge pull request #355 from emulsify-ds/release-7

Release 7

Commit:
fix(theme): fall back to stored favicon svg

callinmullaney committed:
Merge pull request #356 from emulsify-ds/release-7

fix(theme): fall back to stored favicon svg

Commit:
fix(theme): mark favicon previews as safe markup

Commit:
chore(release): align theme package metadata

Commit:
fix(theme): clean favicon settings persistence

Commit:
chore: reformat add_attributes being set in twig templates

Commit:
fix(theme): align add_attributes with helper semantics

callinmullaney committed:
Merge pull request #357 from emulsify-ds/codex-add-attributes-helper-cleanup

fix(theme): align add_attributes with helper semantics

Commit:
feat: refine git actions tests

callinmullaney committed:
Merge pull request #358 from emulsify-ds/codex-add-attributes-helper-cleanup

feat: refine git actions tests

Commit:
fix: refine render ref pages test

Commit:
fix: adjust favicon generation test

Commit:
refactor(theme): use bem helper for static BEM classes

Commit:
fix(ci): prefer env theme in favicon smoke

Commit:
chore: harden 7.x theme readiness

Commit:
fix(ci): normalize favicon reset smoke defaults

Commit:
chore: remove unused Environment and deployment status section of the theme

Commit:
docs: add 7.x upgrade and template parity guides

Commit:
chore: code comments

Commit:
feat: set base class as a variable for some templates

Commit:
feat: add Drupal 12 theme compatibility

Commit:
ci: add Drupal 12 dev readiness coverage

Commit:
Merge remote-tracking branch 'origin/main' into 7.x

* origin/main:
  docs: clarify readiness scripts
  chore: defer starterkit scope to 7.x
  docs: add 7.x preparation guide
  chore: minimize release tooling churn
  feat: add views display type template suggestion
  fix: type field hook suggestion argument
  test: cover current Drupal minors in readiness workflow
  test: add baseline template parity check
  fix: align baseline templates with stable9 parity
  fix: keep starterkit generation out of 6.x release
  fix: refine hook class migration
  feat: project dependency updates
  docs: comment theme readiness scripts
  fix: normalize stable9-only form classes
  fix: harden 6.x theme readiness checks

# Conflicts:
#	.browserslistrc
#	.github/scripts/render-parity.sh
#	.github/scripts/render-reference-pages.sh
#	.github/scripts/setup-fixture-site.sh
#	.github/scripts/smoke-without-stable9.sh
#	.github/scripts/template-parity.sh
#	.github/workflows/theme-readiness.yml
#	README.md
#	includes/field.inc
#	package.json
#	src/Hook/FieldHooks.php
#	src/Hook/FormHooks.php
#	src/Hook/ViewsHooks.php
#	templates/form/form.html.twig
#	templates/layout/page.html.twig
#	templates/misc/status-messages.html.twig
#	whisk/whisk.info.emulsify.yml
#	whisk/whisk.info.yml

Commit:
ci: make dev readiness non-blocking

Commit:
ci: ignore dev branch readiness step failures

Commit:
test: tighten theme readiness scripts

Commit:
ci: avoid duplicate readiness runs

Commit:
ci: keep release branch readiness coverage

Commit:
refactor!: make favicon generation settings-driven

Move favicon form work into a dedicated class.

Remove runtime fallback generation from page attachments.

Attach only existing favicon packages at runtime.

Add lifecycle docs and smoke coverage for the policy.

Commit:
docs!: polish 7.x release and Starterkit messaging

Clarify Drupal 11.3+ support and Drupal 12 forward compatibility.

Document known breaking changes and favicon deployment ownership.

Use child theme and parent theme language consistently.

Commit:
chore!: harden 7.x release readiness checks

Expand release checks for hooks, favicon settings, and Starterkit output.

Verify generated child themes and upload smoke artifacts on failure.

Refresh template parity language for the parent-theme removal.

Commit:
fix: normalize favicon SVG sources

Commit:
fix: harden favicon head link cleanup

Commit:
chore: add Twig story audit script

Commit:
chore: expose project audit in whisk

Commit:
fix: add dependency audit overrides

Commit:
feat: refine Emulsify Core config templates

Commit:
refactor(a11y): convert Whisk config to ESM

Commit:
fix(release): recognize breaking commit headers

Commit:
fix(theme): preserve stable9 marker attributes

Commit:
chore: remove sync with d.o

Commit:
fix(release): keep drupal sync manual

Commit:
fix(ci): split starterkit smoke phases
